editor/plugins/spatial_editor_plugin.cpp

```diff
@@ -4211,28 +4417,47 @@ SpatialEditorViewport::SpatialEditorViewport(SpatialEditor *p_spatial_editor, Ed
 	previewing_cinema = false;
 
 	locked_label = memnew(Label);
-	locked_label->set_anchor_and_margin(MARGIN_TOP, ANCHOR_END, -20 * EDSCALE);
-	locked_label->set_anchor_and_margin(MARGIN_BOTTOM, ANCHOR_END, -10 * EDSCALE);
-	locked_label->set_h_grow_direction(GROW_DIRECTION_END);
-	locked_label->set_v_grow_direction(GROW_DIRECTION_BEGIN);
 	locked_label->set_align(Label::ALIGN_CENTER);
-	surface->add_child(locked_label);
+	locked_label->set_h_size_flags(SIZE_SHRINK_CENTER);
+	bottom_center_vbox->add_child(locked_label);
 	locked_label->set_text(TTR("View Rotation Locked"));
 	locked_label->hide();
 
 	zoom_limit_label = memnew(Label);
-	zoom_limit_label->set_anchors_and_margins_preset(LayoutPreset::PRESET_BOTTOM_LEFT);
-	zoom_limit_label->set_margin(Margin::MARGIN_TOP, -28 * EDSCALE);
 	zoom_limit_label->set_text(TTR("To zoom further, change the camera's clipping planes (View -> Settings...)"));
 	zoom_limit_label->set_name("ZoomLimitMessageLabel");
 	zoom_limit_label->add_color_override("font_color", Color(1, 1, 1, 1));
 	zoom_limit_label->hide();
-	surface->add_child(zoom_limit_label);
+	bottom_center_vbox->add_child(zoom_limit_label);
 
 	top_right_vbox = memnew(VBoxContainer);
 	top_right_vbox->set_anchors_and_margins_preset(PRESET_TOP_RIGHT, PRESET_MODE_MINSIZE, 2.0 * EDSCALE);
 	top_right_vbox->set_h_grow_direction(GROW_DIRECTION_BEGIN);
 
+#ifdef ANDROID_ENABLED
+	position_control = memnew(ViewportNavigationControl);
+	position_control->set_navigation_mode(SpatialEditorViewport::NAVIGATION_MOVE);
+	position_control->set_custom_minimum_size(Size2(80, 80) * EDSCALE);
+	position_control->set_h_size_flags(SIZE_SHRINK_END);
+	position_control->set_anchor_and_margin(MARGIN_LEFT, ANCHOR_BEGIN, 100 * EDSCALE);
+	position_control->set_anchor_and_margin(MARGIN_TOP, ANCHOR_END, -135 * EDSCALE);
+	position_control->set_anchor_and_margin(MARGIN_RIGHT, ANCHOR_BEGIN, 180 * EDSCALE);
+	position_control->set_anchor_and_margin(MARGIN_BOTTOM, ANCHOR_END, -55 * EDSCALE);
+	position_control->set_viewport(this);
+	surface->add_child(position_control);
+
+	look_control = memnew(ViewportNavigationControl);
+	look_control->set_navigation_mode(SpatialEditorViewport::NAVIGATION_LOOK);
+	look_control->set_custom_minimum_size(Size2(80, 80) * EDSCALE);
+	look_control->set_h_size_flags(SIZE_SHRINK_END);
+	look_control->set_anchor_and_margin(MARGIN_LEFT, ANCHOR_END, -180 * EDSCALE);
+	look_control->set_anchor_and_margin(MARGIN_TOP, ANCHOR_END, -135 * EDSCALE);
+	look_control->set_anchor_and_margin(MARGIN_RIGHT, ANCHOR_END, -100 * EDSCALE);
+	look_control->set_anchor_and_margin(MARGIN_BOTTOM, ANCHOR_END, -55 * EDSCALE);
+	look_control->set_viewport(this);
+	surface->add_child(look_control);
+#endif
```

Comment:
I wonder if we should make this dependent on a bool, that could be overridden via an editor settings, and default to true on Android.

I could imagine this being useful for the Web editor on mobile devices too, or even convertible laptops with touchscreen.

We could also make it dependent on `DisplayServer::get_singleton()->screen_is_touchscreen()`, though we should still have sensible defaults regardless (on by default on Android editor, off by default on Windows/Linux laptop with touchscreen).

Alternatively it could be exposed as a button in the viewport toolbar, which can be on by default on Android and off by default otherwise.

Reply:
Good point, there was a similar comment on twitter about making it available to more platforms than Android, so I'll update the logic to enable that.

In terms of customization, I was thinking about providing the ability to move the controls given differences in hands sizes and screen sizes. Do we have similar capability in the editor I can use for reference?